examples/stable-diffusion/training/train_text_to_image.py

```diff
@@ -0,0 +1,1737 @@
+#!/usr/bin/env python
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Fine-tuning script for Stable Diffusion models for text2image with support for LoRA.
+Adapted from the following sources:
+https://github.com/huggingface/diffusers/blob/v0.23.1/examples/text_to_image/train_text_to_image_sdxl.py
+https://github.com/huggingface/diffusers/blob/v0.23.1/examples/text_to_image/train_text_to_image_lora_sdxl.py
```

Comment:
Maybe this script should be renamed to `train_text_to_image_sdxl.py` as I see that Diffusers also has an example called [`train_text_to_image.py`](https://github.com/huggingface/diffusers/tree/v0.23.1/examples/text_to_image).

Reply:
@regisss  should we keep 1 script and use it to train both stable diffusion and stable diffusion xl with parameter?

Reply:
Ideally yes, if that doesn't make the script too complicated